File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateAdapter.java
package com.neyogiry.android.sample;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class DateAdapter extends RecyclerView.Adapter<DateAdapter.ViewHolder>{

    private ArrayList<String> mDates;

    public DateAdapter(ArrayList<String> dates){
        this.mDates = dates;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        View mView = inflater.inflate(R.layout.item_date, parent, false);
        return new ViewHolder(mView);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String dateTime = mDates.get(position);
        holder.date.setText(dateTime);
    }

    @Override
    public int getItemCount() {
        return mDates.size();
    }

    public void updateList(ArrayList<String> list){
        this.mDates = list;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        private TextView date;

        public ViewHolder(View itemView) {
            super(itemView);
            date = (TextView)itemView.findViewById(R.id.tv_date);
        }
    }

}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateAdapter.kt
package com.neyogiry.android.sample

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.item_date.view.*

class DateAdapter(private var mDates: ArrayList<String>): RecyclerView.Adapter<DateAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, position: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_date, parent, false))
    }

    override fun getItemCount(): Int {
        return mDates.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dateTime = mDates?.get(position)
        holder.dateTime?.text = dateTime
    }

    fun updateList(list: ArrayList<String>) {
        this.mDates = list
        notifyDataSetChanged()
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var dateTime: TextView = itemView?.tv_date
    }

}